win32ss/user/ntuser/ntstubs.c

```diff
@@ -482,6 +482,9 @@ PIMC FASTCALL UserCreateInputContext(ULONG_PTR dwClientImcData)
     if (!pIMC)
         return NULL;
 
+    // Release the extra reference (UserCreateObject added 2 references).
+    UserDereferenceObject(pIMC);
```

Comment:
If you don't plan to add any additional callers that need the pointer, you could have `UserCreateInputContext` return the handle instead of the `PIMC`. That would feel a little cleaner than returning an un-referenced pointer.

Reply:
Shouldn't technically wait until the very end of this `UserCreateInputContext` function to call the dereference? This is what (afaics) all the other functions that use the `UserCreateObject` function do:
1. Call `UserCreateObject`
2. In case of success: allocate extra stuff in the returned object.
3. Finally, call `UserDereferenceObject` on the object to remove the extra references and return.